### PR TITLE
feat(most-visited.sql): new query for site diff

### PR DESCRIPTION
**Pull request by MarquiseRosier**
this creates a query that will select most popular pages and elect one if there are many under one domain. The way one link out of many that share a similar third level domain is elected, is that I just don't consider those currently. We only pick the distinct URLS where just one of them has the max value; I think this suffices for getting started but should be fixed!

@kptdobe @stefan-guggisberg if you guys don't mind helping out with any advice on it that'd be great :D 

Please ensure your pull request adheres to the following guidelines:
- [ ] make sure to link the related issues in this description
- [ ] when merging / squashing, make sure the fixed issue references are visible in the commits, for easy compilation of release notes

## Related Issues
#200 

Thanks for contributing!

### CODE REVIEW

File: src/queries/most-visited.sql [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+--- limit: 100
+with topurls as ( SELECT 
+    req_http_X_Owner AS owner, 
+    req_http_X_Repo AS repo,
+    CONCAT(req_http_X_Owner, req_http_X_Repo) as name,
+    req_url,
+    COUNT(time_start_usec) AS requests,
+  FROM `helix_logging_1McGRQOYFuABWBHyD8D4Ux.requests*`
+  WHERE 
+    resp_http_Content_Type LIKE "%html%" AND
```
Comment (kptdobe) [inline src/queries/most-visited.sql:14]: `like '%xyz%'` is usually "expensive" for a query. Are there so many combinations that we cannot be more specific ?

File: src/queries/most-visited.sql [hunk 2]
```diff
@@ -0,0 +1,45 @@
+--- Authorization: none
+--- Cache-Control: max-age=1296000
+--- fromDays: 30
+--- toDays: 0
+--- limit: 100
```
Comment (trieloff) [inline src/queries/most-visited.sql:5]: Better default: 30 days.

File: src/queries/most-visited.sql (hunk 1 above)
Comment (trieloff) [reply src/queries/most-visited.sql:14]: We could drill it down to `%text/html%`, but there may be a charset appendix.

**Comment by kptdobe** [comment]
Looking at https://adobeioruntime.net/api/v1/web/helix/helix-services/run-query@ci1100/most-visited?limit=20, I think it would be useful to add some data like number of visits. Otherwise this list of urls is a bit useless.

**Comment by stefan-guggisberg** [comment]
> Looking at https://adobeioruntime.net/api/v1/web/helix/helix-services/run-query@ci1100/most-visited?limit=20, I think it would be useful to add some data like number of visits. Otherwise this list of urls is a bit useless.

I agree. See https://github.com/adobe/helix-run-query/pull/201#pullrequestreview-417765155. The proposed query returns `num_reqs`, `repo_name` and `req_url`

**Comment by MarquiseRosier** [comment]
> I agree. See [#201 (review)](https://github.com/adobe/helix-run-query/pull/201#pullrequestreview-417765155). The proposed query returns `max_reqs`, `repo_name` and `req_url`

Thanks @stefan-guggisberg for the fix; that is amazingly fast. The only thing is that there are duplicates in the result set; I counted 18; Is there a better way to get rid of duplicates aside from running this hideous snippet of code; I can find a better way; just wondering if you have something up your sleeve. 
```
        select req_url, requests 
        from urls d 
        where ( 
          select (
            select count(name) as count 
            from urls 
            where d.name = name group by name)) = 1 
        order by requests desc limit 100)
```
If not I think that final step is necessary to do the final sweep and remove the duplicates! That might slow down your query a bit;  I would rather waste Google's time processing especially cause there's no change in the MBs processed in both queries. I don't want to waste calls to our visual validation resource by testing the same site x times more than is necessary. 

Also @kptdobe 
I'm struggling to see why just serving the urls makes it useless? Don't we plan to use the query for visual validation testing; and in that case we just need to determine what sites are most important to test

**Comment by stefan-guggisberg** [comment]
> The only thing is that there are duplicates in the result set; 

The _duplicates_ aren't really duplicates. These are cases where a site has multiple urls all sharing the exact same maximum number of visits. The likelihood of such cases increases with very low numbers of visits (-> insignificant sites). If you look at the results you'll see that the _duplicates_  all have counts of less than 20 visits (in 30 days). We wouldn't want to waste calls to our visual validation resource by testing sites with so few visits, right? If we apply a reasonable threshold (e.g. more than 100 visits in 30 days) we will hardly ever see _duplicates_.

Avoiding _duplicates_ is certainly possible but IMO not worth the implied cost (complexity/ugliness of query and increased resource consumption).  I would consider it a classical case of premature optimisation.

**Comment by stefan-guggisberg** [comment]
> I'm struggling to see why just serving the urls makes it useless? We plan to use the query for visual validation testing; if so than we just need to determine what sites are most important to test

Just returning urls ignores the relevance (number of visits) of the result, see my previous comment.

**Comment by MarquiseRosier** [comment]
@stefan-guggisberg good points, i'll update the PR to limit results to those with > 100 reqs and commit your code; thanks a ton for this help!

File: src/queries/most-visited.sql [hunk 3]
```diff
@@ -0,0 +1,37 @@
+--- Authorization: none
```
Comment (tripodsan) [inline src/queries/most-visited.sql:1]: can we add a copyright header here?

**Pull request merged**